R/expand.R

```diff
@@ -193,23 +220,29 @@ expand_approximate_months <- function(dates, approx_range) {
   mr <- 30.42 * approx_range
   dates <- lapply(dates, function(x) {
     # One Month
-    x <- ifelse(approx_range == 1 & stringr::str_detect(x, "-\\~04-|-\\~06-|-\\~09-|-\\~11-"),
+    x <- ifelse(approx_range == 1 &
+                  stringr::str_detect(x, "-\\~04-|-\\~06-|-\\~09-|-\\~11-"),
                 paste0(as.Date(gsub("\\~", "", x)) - 31, "..",
                        as.Date(gsub("\\~", "", x)) + 30), x)
-    x <- ifelse(approx_range == 1 & stringr::str_detect(x, "-\\~05-|-\\~10-|-\\~12-|-\\~07-"),
+    x <- ifelse(approx_range == 1 &
+                  stringr::str_detect(x, "-\\~05-|-\\~10-|-\\~12-|-\\~07-"),
                 paste0(as.Date(gsub("\\~", "", x)) - 30, "..",
                        as.Date(gsub("\\~", "", x)) + 31), x)
-    x <- ifelse(approx_range == 1 & stringr::str_detect(x, "-\\~01-|-\\~08-"),
+    x <- ifelse(approx_range == 1 &
+                  stringr::str_detect(x, "-\\~01-|-\\~08-"),
                 paste0(as.Date(gsub("\\~", "", x)) - 31, "..",
                        as.Date(gsub("\\~", "", x)) + 31), x)
-    x <- ifelse(approx_range == 1 & stringr::str_detect(x, "-\\~03-") & # leap year
+    x <- ifelse(approx_range == 1 &
+                  stringr::str_detect(x, "-\\~03-") & # leap year
                   lubridate::leap_year(lubridate::as_date(gsub("\\~", "", x))),
                 paste0(as.Date(gsub("\\~", "", x)) - 29, "..",
```

Comment:
I wonder whether this could be it's own user-facing function?

Reply:
Do you mean the approximate year and months date expansion could become user-facing? I can make these user-facing if you would like, just let me know, but I do not know if I see when users would use such instead of simply running `expand()`. 

I also have been thinking about the "approx_range" argument we use to expand dates where on_before/on_after are present. We use, arbitrarily, 3 years for this expansion. @jhollway should we make this either more interactive, or more explicit by adding a warning message, for users so that they can select if they want to expand these dates and by how much?